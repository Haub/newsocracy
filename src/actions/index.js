export const addCategoryNews = (articles) => ({
  type: 'ADD_CATEGORY_NEWS',
  articles
});

