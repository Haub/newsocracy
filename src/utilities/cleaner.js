export const cleanLeftData = (article, category, index) => {
  const date = article.publishedAt.slice(0,10);

  return {
    source: article.source,
    title: article.title,
    date: date,
    author: article.author,
    url: article.url,
    image: article.urlToImage,
    content: article.content,
    category: category,
    id: 'left' + index
  }
};

export const cleanCenterData = (article, category, index) => {
  const date = article.publishedAt.slice(0,10);

  return {
    source: article.source,
    title: article.title,
    date: date,
    author: article.author,
    url: article.url,
    image: article.urlToImage,
    content: article.content,
    category: category,
    id: 'center' + index
  }
};

export const cleanRightData = (article, category, index) => {
  const date = article.publishedAt.slice(0,10);
  
  return {
    source: article.source,
    title: article.title,
    date: date,
    author: article.author,
    url: article.url,
    image: article.urlToImage,
    content: article.content,
    category: category,
    id: 'right' + index
  }
};