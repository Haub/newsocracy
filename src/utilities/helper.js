import { key } from './apiKey';

export const getNews = async() => {
  const response = await fetch(`https://newsapi.org/v2/everything?q=metoo&domains= breitbart.com,  theamericanconservative.com&from=2018-10-22&to=2018-10-28&sortBy=popularity&apiKey=${key}`);
  const data = await response.json();
  console.log(data);

}