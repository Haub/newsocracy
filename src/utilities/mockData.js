export const article = {
  source: 'npr.org',
  title: 'Immigration',
  date: '2018-10-31',
  author: 'Joe Smith',
  url: 'www.npr.org',
  image: 'www.npr.org/image'
}