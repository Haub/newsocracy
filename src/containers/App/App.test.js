import React from 'react';
import { shallow } from 'enzyme';
import { App } from './index.js';
import { mapStateToProps, mapDispatchToProps } from './index.js';
import { addCategoryNews } from '../../actions';
import * as API from '../../utilities/helper';


describe('APP', () => { 
  it('should match the snapshot', () => {
   const wrapper = shallow(<App />);
   expect(wrapper).toMatchSnapshot();
  });

  it('should invoke getNews when a button is pressed', () => {
    const wrapper = shallow(<App />);
    const spy = jest.spyOn(API, 'getNews');
    const mockArticle = {
      source: 'MSNBC',
      title: 'Election',
      publishedAt: '2018-11-05T15:47:25Z',
      author: 'Joe',
      url: 'www.cnn.com',
      category: 'left',
    };
    wrapper.instance().handleClick('immigration');
    expect(spy).toHaveBeenCalled();
  });
})

describe('mapStateToProps', () => {
  it('should have access to the articles array', () => {
    const mockStore = {
      articles: []
    };
    const expected = {...mockStore};
    const result = mapStateToProps(mockStore);
    expect(result).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  it('should add articles to the store with addCategoryNews', () => {
    const mockDispatch = jest.fn();
    const actiontoDispatch = addCategoryNews([]);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.addCategoryNews([]);
    expect(mockDispatch).toHaveBeenCalledWith(actiontoDispatch);
  });
})