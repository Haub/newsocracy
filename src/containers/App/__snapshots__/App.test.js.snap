// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
        >
          <h1
            className="title"
          >
             news-ocracy
          </h1>
          <h5
            className="sub-title"
          >
             seek perspective.  identify bias.  engage in civil discourse. 
          </h5>
        </header>,
        <section
          className="button-container"
        >
          <NavLink
            activeClassName="selected"
            aria-current="page"
            to="/immigration"
          >
            <button
              className="topic-button"
              name="immigration"
              onClick={[Function]}
            >
              Immigration
            </button>
          </NavLink>
          <NavLink
            activeClassName="selected"
            aria-current="page"
            to="/climate"
          >
            <button
              className="topic-button"
              onClick={[Function]}
            >
              Climate Change
            </button>
          </NavLink>
          <NavLink
            activeClassName="selected"
            aria-current="page"
            to="/midterms"
          >
            <button
              className="topic-button"
              onClick={[Function]}
            >
              Midterm Elections
            </button>
          </NavLink>
          <NavLink
            activeClassName="selected"
            aria-current="page"
            to="/healthcare"
          >
            <button
              className="topic-button"
              onClick={[Function]}
            >
              Healthcare
            </button>
          </NavLink>
          <NavLink
            activeClassName="selected"
            aria-current="page"
            to="/refugees"
          >
            <button
              className="topic-button"
              onClick={[Function]}
            >
              Refugees
            </button>
          </NavLink>
        </section>,
        <Switch>
          <Route
            path="/immigration/:id"
            render={[Function]}
          />
          <Route
            path="/climate/:id"
            render={[Function]}
          />
          <Route
            path="/midterms/:id"
            render={[Function]}
          />
          <Route
            path="/healthcare/:id"
            render={[Function]}
          />
          <Route
            path="/refugees/:id"
            render={[Function]}
          />
          <Route
            path="/immigration"
            render={[Function]}
          />
          <Route
            path="/climate"
            render={[Function]}
          />
          <Route
            path="/midterms"
            render={[Function]}
          />
          <Route
            path="/healthcare"
            render={[Function]}
          />
          <Route
            path="/refugees"
            render={[Function]}
          />
          <Redirect
            push={false}
            to="/"
          />
        </Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="title"
            >
               news-ocracy
            </h1>,
            <h5
              className="sub-title"
            >
               seek perspective.  identify bias.  engage in civil discourse. 
            </h5>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " news-ocracy",
              "className": "title",
            },
            "ref": null,
            "rendered": " news-ocracy",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " seek perspective.  identify bias.  engage in civil discourse. ",
              "className": "sub-title",
            },
            "ref": null,
            "rendered": " seek perspective.  identify bias.  engage in civil discourse. ",
            "type": "h5",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="selected"
              aria-current="page"
              to="/immigration"
            >
              <button
                className="topic-button"
                name="immigration"
                onClick={[Function]}
              >
                Immigration
              </button>
            </NavLink>,
            <NavLink
              activeClassName="selected"
              aria-current="page"
              to="/climate"
            >
              <button
                className="topic-button"
                onClick={[Function]}
              >
                Climate Change
              </button>
            </NavLink>,
            <NavLink
              activeClassName="selected"
              aria-current="page"
              to="/midterms"
            >
              <button
                className="topic-button"
                onClick={[Function]}
              >
                Midterm Elections
              </button>
            </NavLink>,
            <NavLink
              activeClassName="selected"
              aria-current="page"
              to="/healthcare"
            >
              <button
                className="topic-button"
                onClick={[Function]}
              >
                Healthcare
              </button>
            </NavLink>,
            <NavLink
              activeClassName="selected"
              aria-current="page"
              to="/refugees"
            >
              <button
                className="topic-button"
                onClick={[Function]}
              >
                Refugees
              </button>
            </NavLink>,
          ],
          "className": "button-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "selected",
              "aria-current": "page",
              "children": <button
                className="topic-button"
                name="immigration"
                onClick={[Function]}
              >
                Immigration
              </button>,
              "to": "/immigration",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Immigration",
                "className": "topic-button",
                "name": "immigration",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Immigration",
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "selected",
              "aria-current": "page",
              "children": <button
                className="topic-button"
                onClick={[Function]}
              >
                Climate Change
              </button>,
              "to": "/climate",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Climate Change",
                "className": "topic-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Climate Change",
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "selected",
              "aria-current": "page",
              "children": <button
                className="topic-button"
                onClick={[Function]}
              >
                Midterm Elections
              </button>,
              "to": "/midterms",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Midterm Elections",
                "className": "topic-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Midterm Elections",
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "selected",
              "aria-current": "page",
              "children": <button
                className="topic-button"
                onClick={[Function]}
              >
                Healthcare
              </button>,
              "to": "/healthcare",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Healthcare",
                "className": "topic-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Healthcare",
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "selected",
              "aria-current": "page",
              "children": <button
                className="topic-button"
                onClick={[Function]}
              >
                Refugees
              </button>,
              "to": "/refugees",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Refugees",
                "className": "topic-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Refugees",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              path="/immigration/:id"
              render={[Function]}
            />,
            <Route
              path="/climate/:id"
              render={[Function]}
            />,
            <Route
              path="/midterms/:id"
              render={[Function]}
            />,
            <Route
              path="/healthcare/:id"
              render={[Function]}
            />,
            <Route
              path="/refugees/:id"
              render={[Function]}
            />,
            <Route
              path="/immigration"
              render={[Function]}
            />,
            <Route
              path="/climate"
              render={[Function]}
            />,
            <Route
              path="/midterms"
              render={[Function]}
            />,
            <Route
              path="/healthcare"
              render={[Function]}
            />,
            <Route
              path="/refugees"
              render={[Function]}
            />,
            <Redirect
              push={false}
              to="/"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/immigration/:id",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/climate/:id",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/midterms/:id",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/healthcare/:id",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/refugees/:id",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/immigration",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/climate",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/midterms",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/healthcare",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/refugees",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "push": false,
              "to": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
          >
            <h1
              className="title"
            >
               news-ocracy
            </h1>
            <h5
              className="sub-title"
            >
               seek perspective.  identify bias.  engage in civil discourse. 
            </h5>
          </header>,
          <section
            className="button-container"
          >
            <NavLink
              activeClassName="selected"
              aria-current="page"
              to="/immigration"
            >
              <button
                className="topic-button"
                name="immigration"
                onClick={[Function]}
              >
                Immigration
              </button>
            </NavLink>
            <NavLink
              activeClassName="selected"
              aria-current="page"
              to="/climate"
            >
              <button
                className="topic-button"
                onClick={[Function]}
              >
                Climate Change
              </button>
            </NavLink>
            <NavLink
              activeClassName="selected"
              aria-current="page"
              to="/midterms"
            >
              <button
                className="topic-button"
                onClick={[Function]}
              >
                Midterm Elections
              </button>
            </NavLink>
            <NavLink
              activeClassName="selected"
              aria-current="page"
              to="/healthcare"
            >
              <button
                className="topic-button"
                onClick={[Function]}
              >
                Healthcare
              </button>
            </NavLink>
            <NavLink
              activeClassName="selected"
              aria-current="page"
              to="/refugees"
            >
              <button
                className="topic-button"
                onClick={[Function]}
              >
                Refugees
              </button>
            </NavLink>
          </section>,
          <Switch>
            <Route
              path="/immigration/:id"
              render={[Function]}
            />
            <Route
              path="/climate/:id"
              render={[Function]}
            />
            <Route
              path="/midterms/:id"
              render={[Function]}
            />
            <Route
              path="/healthcare/:id"
              render={[Function]}
            />
            <Route
              path="/refugees/:id"
              render={[Function]}
            />
            <Route
              path="/immigration"
              render={[Function]}
            />
            <Route
              path="/climate"
              render={[Function]}
            />
            <Route
              path="/midterms"
              render={[Function]}
            />
            <Route
              path="/healthcare"
              render={[Function]}
            />
            <Route
              path="/refugees"
              render={[Function]}
            />
            <Redirect
              push={false}
              to="/"
            />
          </Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="title"
              >
                 news-ocracy
              </h1>,
              <h5
                className="sub-title"
              >
                 seek perspective.  identify bias.  engage in civil discourse. 
              </h5>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " news-ocracy",
                "className": "title",
              },
              "ref": null,
              "rendered": " news-ocracy",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " seek perspective.  identify bias.  engage in civil discourse. ",
                "className": "sub-title",
              },
              "ref": null,
              "rendered": " seek perspective.  identify bias.  engage in civil discourse. ",
              "type": "h5",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="selected"
                aria-current="page"
                to="/immigration"
              >
                <button
                  className="topic-button"
                  name="immigration"
                  onClick={[Function]}
                >
                  Immigration
                </button>
              </NavLink>,
              <NavLink
                activeClassName="selected"
                aria-current="page"
                to="/climate"
              >
                <button
                  className="topic-button"
                  onClick={[Function]}
                >
                  Climate Change
                </button>
              </NavLink>,
              <NavLink
                activeClassName="selected"
                aria-current="page"
                to="/midterms"
              >
                <button
                  className="topic-button"
                  onClick={[Function]}
                >
                  Midterm Elections
                </button>
              </NavLink>,
              <NavLink
                activeClassName="selected"
                aria-current="page"
                to="/healthcare"
              >
                <button
                  className="topic-button"
                  onClick={[Function]}
                >
                  Healthcare
                </button>
              </NavLink>,
              <NavLink
                activeClassName="selected"
                aria-current="page"
                to="/refugees"
              >
                <button
                  className="topic-button"
                  onClick={[Function]}
                >
                  Refugees
                </button>
              </NavLink>,
            ],
            "className": "button-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "selected",
                "aria-current": "page",
                "children": <button
                  className="topic-button"
                  name="immigration"
                  onClick={[Function]}
                >
                  Immigration
                </button>,
                "to": "/immigration",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Immigration",
                  "className": "topic-button",
                  "name": "immigration",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Immigration",
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "selected",
                "aria-current": "page",
                "children": <button
                  className="topic-button"
                  onClick={[Function]}
                >
                  Climate Change
                </button>,
                "to": "/climate",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Climate Change",
                  "className": "topic-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Climate Change",
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "selected",
                "aria-current": "page",
                "children": <button
                  className="topic-button"
                  onClick={[Function]}
                >
                  Midterm Elections
                </button>,
                "to": "/midterms",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Midterm Elections",
                  "className": "topic-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Midterm Elections",
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "selected",
                "aria-current": "page",
                "children": <button
                  className="topic-button"
                  onClick={[Function]}
                >
                  Healthcare
                </button>,
                "to": "/healthcare",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Healthcare",
                  "className": "topic-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Healthcare",
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "selected",
                "aria-current": "page",
                "children": <button
                  className="topic-button"
                  onClick={[Function]}
                >
                  Refugees
                </button>,
                "to": "/refugees",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Refugees",
                  "className": "topic-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Refugees",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                path="/immigration/:id"
                render={[Function]}
              />,
              <Route
                path="/climate/:id"
                render={[Function]}
              />,
              <Route
                path="/midterms/:id"
                render={[Function]}
              />,
              <Route
                path="/healthcare/:id"
                render={[Function]}
              />,
              <Route
                path="/refugees/:id"
                render={[Function]}
              />,
              <Route
                path="/immigration"
                render={[Function]}
              />,
              <Route
                path="/climate"
                render={[Function]}
              />,
              <Route
                path="/midterms"
                render={[Function]}
              />,
              <Route
                path="/healthcare"
                render={[Function]}
              />,
              <Route
                path="/refugees"
                render={[Function]}
              />,
              <Redirect
                push={false}
                to="/"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/immigration/:id",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/climate/:id",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/midterms/:id",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/healthcare/:id",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/refugees/:id",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/immigration",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/climate",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/midterms",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/healthcare",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/refugees",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "push": false,
                "to": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
